apps/veyfi/components/RedeemTab.tsx

```diff
@@ -29,18 +29,49 @@ function RedeemTab(): ReactElement {
 	const {refresh: refreshBalances} = useWallet();
 	const {getRequiredEth, price: optionPrice, positions, allowances, isLoading: isLoadingOption, refresh} = useOption();
 	const clearLockAmount = (): void => set_redeemAmount(toNormalizedBN(0));
-	const refreshData = (): unknown => Promise.all([refresh(), refreshBalances()]);
-	const onTxSuccess = (): unknown => Promise.all([refreshData(), clearLockAmount()]);
+	const refreshData = useCallback((): unknown => Promise.all([refresh(), refreshBalances()]), [refresh, refreshBalances]);
+	const onTxSuccess = useCallback((): unknown => Promise.all([refreshData(), clearLockAmount()]), [refreshData]);
 	const [{status, result}, fetchRequiredEth] = useAsync(getRequiredEth, BIG_ZERO);
 	const ethBalance = useBalance(ETH_TOKEN_ADDRESS);
 	const yfiPrice = useTokenPrice(YFI_ADDRESS);
-	const [approveRedeem, approveRedeemStatus] = useTransaction(approveERC20, refreshData);
-	const [redeem, redeemStatus] = useTransaction(OptionActions.redeem, onTxSuccess);
+	const [approveRedeemStatus, set_approveRedeemStatus] = useState(defaultTxStatus);
+	const [redeemStatus, set_redeemStatus] = useState(defaultTxStatus);
 
 	const userAddress = address as TAddress;
 	const oYFIBalance = toNormalizedBN(formatBigNumberAsAmount(positions?.balance), 18);
 	const ethRequired = toNormalizedValue(result, 18);
 
+	const handleApproveRedeem = useCallback(async (): Promise<void> => {
+		const response = await approveERC20({
+			connector: provider,
+			contractAddress: VEYFI_OYFI_ADDRESS,
+			spenderAddress: VEYFI_OPTIONS_ADDRESS,
+			statusHandler: set_approveRedeemStatus,
+			amount: redeemAmount.raw
```

Comment:
Unsure about this amount

Reply:
All good then, I'm lacking a bit of context. It would be great to have @xgambitox confirm this too :)